include/tss2/tss2_tcti_spi.h

```diff
@@ -0,0 +1,45 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright 2020 Fraunhofer SIT. All rights reserved.
+ */
+#ifndef TSS2_TCTI_SPI_H
+#define TSS2_TCTI_SPI_H
+
+#include "tss2_tcti.h"
+#include <driver/spi_master.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define TSS2_TCTI_SPI_PLATFORM_MAGIC 0x61F7D74109DB0D8BULL
+
+typedef struct TSS2_TCTI_SPI_PLATFORM TSS2_TCTI_SPI_PLATFORM;
+typedef void (*TSS2_TCTI_SPI_PLATFORM_SLEEP_MS_FUNC) (TSS2_TCTI_SPI_PLATFORM* platform, int milliseconds);
```

Comment:
IMHO, this is a bit ugly. Instead of platform, we should only pass the user_data pointer to the callback functions.
Also applies to all subsequent function defs.

Question is, if all callbacks share one userdata or if we want to have a different userdata pointer for each callback ?

Reply:
We now only pass user_data to the callback functions.

In most cases the user_data will contain a struct with SPI device handle, SPI CS pin handle, timeout expiry time.

A shared user_data pointer is needed for e.g. the timeout functions which need to operate on the same timer / timestamp. I think different user data pointers per method would only be confusing and not helpful.